tests/advanced_visibility.go

```diff
@@ -2649,6 +2652,96 @@ func (s *AdvancedVisibilitySuite) TestBuildIdScavenger_DeletesUnusedBuildId() {
 	s.Require().Equal(0, len(res.BuildIdReachability[0].TaskQueueReachability))
 }
 
+func (s *AdvancedVisibilitySuite) TestScheduleListingWithSearchAttributes() {
+	ctx := NewContext()
+
+	// Test 1: List schedule with "scheduleId" query
+	scheduleID := "test-schedule-" + uuid.New()
+	workflowType := "test-workflow-type"
+	workflowID := "test-schedule-" + uuid.New()
+
+	schedule := &workflowservice.CreateScheduleRequest{
+		Namespace:  s.namespace,
+		RequestId:  uuid.New(),
+		ScheduleId: scheduleID,
+		Schedule: &schedulepb.Schedule{
+			Action: &schedulepb.ScheduleAction{
+				Action: &schedulepb.ScheduleAction_StartWorkflow{
+					StartWorkflow: &workflowpb.NewWorkflowExecutionInfo{
+						WorkflowId:   workflowID,
+						WorkflowType: &commonpb.WorkflowType{Name: workflowType},
+						TaskQueue:    &taskqueuepb.TaskQueue{Name: "default"},
+					},
+				},
+			},
+			Policies: &schedulepb.SchedulePolicies{},
+			Spec: &schedulepb.ScheduleSpec{
+				Interval: []*schedulepb.IntervalSpec{
+					{Interval: durationpb.New(time.Hour)},
+				},
+			},
+		},
+	}
+
+	_, err := s.client.CreateSchedule(ctx, schedule)
+	s.NoError(err)
+
+	listRequest := &workflowservice.ListSchedulesRequest{
+		Namespace:       s.namespace,
+		MaximumPageSize: 1,
+		Query:           fmt.Sprintf(`%s = "%s"`, searchattribute.ScheduleID, scheduleID),
+	}
+
+	s.Eventually(func() bool {
+		listResponse, err := s.client.ListSchedules(ctx, listRequest)
+		if err != nil || len(listResponse.Schedules) != 1 {
+			return false
+		}
+
+		return listResponse.Schedules[0].ScheduleId == scheduleID
+	}, 30*time.Second, 1*time.Second)
+
+	listRequest.Query = fmt.Sprintf(`%s IN ("%s")`, searchattribute.ScheduleID, scheduleID)
```

Comment:
Can you add some more value inside the `IN` tuple to test multiple values? It can be anything for this case, it won't matter.